clang/test/SemaSYCL/unique_stable_name.cpp

```diff
@@ -0,0 +1,174 @@
+// RUN: %clang_cc1 %s --std=c++17 -fsyntax-only -fsycl-is-device -Wno-unused -verify
+
+template <typename KernelName, typename KernelType>
+[[clang::sycl_kernel]] void kernel_single_task(KernelType kernelFunc) { // #1 #2 #3 #4 #5 #6 #7 #8 #9 #10 #11
+  kernelFunc();
+}
+
+template <typename Func>
+void kernel3func(const Func &F3) {
+  constexpr const char *F3_output = __builtin_unique_stable_name(F3);
+  // expected-error@#8{{kernel instantiation changes the result of an evaluated '__builtin_unique_stable_name'}}
+  // expected-note@+1{{in instantiation of function template specialization}}
+  kernel_single_task<class kernel3>(F3); // 8th instantiation throwing error
+}
+
+template <typename Func>
+void kernel4func(const Func &F4) {
+  constexpr const char *F4_output = __builtin_unique_stable_name(F4);
+  // expected-error@#9{{kernel instantiation changes the result of an evaluated '__builtin_unique_stable_name'}}
+  // expected-note@+1{{in instantiation of function template specialization}}
+  kernel_single_task<class kernel4>([]() {}); // 9th instantiation throwing error
+}
+
+template <template <typename> typename Outer, typename Inner>
+struct S {
+  void operator()() const;
+};
+
+template <typename Ty>
+struct T {};
+
+template <typename Func>
+void kernel13_14func(const Func &F) {
+  // expected-error@#10{{kernel instantiation changes the result of an evaluated '__builtin_unique_stable_name'}}
+  // expected-note@+1{{in instantiation of function template specialization}}
+  kernel_single_task<class kernel13>(F); // 10th instantiation throwing error
+  // expected-error@#11{{kernel instantiation changes the result of an evaluated '__builtin_unique_stable_name'}}
+  // expected-note@+1{{in instantiation of function template specialization}}
+  kernel_single_task<class kernel14>(F); // 11th instantiation throwing error
+  // Using the same functor twice should be fine
+}
+
+template <typename Ty>
+auto func() -> decltype(__builtin_unique_stable_name(Ty::str));
+
+struct Derp {
+  static constexpr const char str[] = "derp derp derp";
+};
+
+template <typename T>
+static constexpr const char *output1 = __builtin_unique_stable_name(T);
+
+#define MACRO12()                \
+  auto l12 = []() { return 1; }; \
+  constexpr const char *l1_output = __builtin_unique_stable_name(l12);
+
+int main() {
+
+  // kernel0
+  // kernel_single_task<class kernel0>(func<Derp>());
```

Comment:
@schittir -- is there a reason this one is still commented out?

Reply:
No reason. Uncommented.